engine/src/object.h

```diff
@@ -1373,4 +1386,131 @@ T* MCObjectCast(MCObject* p_object)
     return static_cast<T*> (p_object);
 }
 
+
+// Automatic (RAII) class for storage of object handles
+class MCObjectProxy::Handle
+{
+public:
+    
+    Handle() :
+    m_proxy(NULL)
+    {
+    }
+    
+    Handle(MCObjectProxy* p_proxy)
+    : m_proxy(NULL)
+    {
+        Set(p_proxy);
+    }
+    
+    Handle(const Handle& p_handle) :
+    m_proxy(NULL)
+    {
+        Set(p_handle.m_proxy);
+    }
+    
+    Handle& operator= (MCObjectProxy* p_proxy)
+    {
+        Set(p_proxy);
+        return *this;
+    }
+    
+    Handle& operator= (const Handle& p_handle)
+    {
+        Set(p_handle.m_proxy);
+        return *this;
+    }
+    
+    ~Handle()
+    {
+        if (m_proxy)
```

Comment:
In our coding style we currently compare pointer values explicitly against `nil`.